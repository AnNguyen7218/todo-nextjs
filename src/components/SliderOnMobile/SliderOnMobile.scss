@import 'config.media';
@import 'config.gutters';
@import 'config.colors';
@import 'config.fonts';

.mobile-container {
  display: block;

  @include break(md) {
    display: none;
  }
}

.desktop-container {
  display: none;

  @include break(md) {
    display: block;
  }
}

.progress-bar {
  bottom: 0;
  top: unset !important;
  height: 0.2rem !important;
}

.progressbar-fill {
  background: $black;
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  transform: scale(0);
  transform-origin: left top;
}

.count-slides {
  @include small;
  margin-top: gutter(8);
}
