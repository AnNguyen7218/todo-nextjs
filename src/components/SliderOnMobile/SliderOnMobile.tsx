/* eslint-disable @typescript-eslint/no-explicit-any */
import React, { useState } from 'react';
import {
  ReactIdSwiperProps,
  ReactIdSwiperChildren,
  GetSwiper,
} from 'react-id-swiper/lib/types';
import cx from 'classnames';
import { Container, Slider } from 'components';
import css from './SliderOnMobile.scss';

type Props = {
  desktopComponent: React.ReactNode;
  mobileComponent: ReactIdSwiperChildren;
  desktopContainerClassName?: string;
  mobileContainerClassName?: string;
  swiperParams?: ReactIdSwiperProps;
  getSwiper?: GetSwiper;
};

const SliderOnMobile: React.FC<Props> = ({
  desktopComponent,
  mobileComponent,
  swiperParams = {},
  mobileContainerClassName,
  desktopContainerClassName,
  getSwiper,
}) => {
  const [current, setCurrent] = useState(1);
  const [total, setTotal] = useState(0);

  const defaultSwiperPrams: ReactIdSwiperProps = {
    loop: false,
    pagination: {
      el: `.swiper-pagination.swiper-pagination-progressbar.${css['progress-bar']}`,
      type: 'progressbar',
      progressbarFillClass: `${css['progressbar-fill']}`,
    },
    on: {
      init: function() {
        const swiper: any = this;
        setTotal(swiper.slides.length);
      },
      slideChange: function() {
        const swiper: any = this;
        setCurrent(swiper.realIndex + 1);
      },
      resize: function() {
        const swiper: any = this;
        setTotal(swiper.slides.length);
        swiper.updateSize();
      },
    },
  };

  const customSwiperParams = {
    ...defaultSwiperPrams,
    ...swiperParams,
  };

  const currentPagination = current.toString().padStart(2, '0');
  const totalSlides = total.toString().padStart(2, '0');

  return (
    <>
      <Container
        className={cx(css['desktop-container'], desktopContainerClassName)}
      >
        {desktopComponent}
      </Container>
      <Container
        className={cx(css['mobile-container'], mobileContainerClassName)}
      >
        <Slider swiperParams={customSwiperParams} getSwiper={getSwiper}>
          {mobileComponent}
        </Slider>
        <p className={css['count-slides']}>
          {currentPagination} / {totalSlides}
        </p>
      </Container>
    </>
  );
};

export default SliderOnMobile;
