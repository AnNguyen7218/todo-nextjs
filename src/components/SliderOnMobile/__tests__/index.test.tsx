import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SliderOnMobile from '../SliderOnMobile';

afterEach(cleanup);

const ITEMS = ['item1', 'item2', 'item3', 'item4'];

describe('SliderOnMobile Component', () => {
  test('SliderOnMobile should match snapshot', () => {
    const renderContent = () => {
      return ITEMS.map((item, index) => {
        return <div key={index}>{item}</div>;
      });
    };
    const { container } = render(
      <SliderOnMobile
        desktopComponent={renderContent()}
        mobileComponent={renderContent()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
