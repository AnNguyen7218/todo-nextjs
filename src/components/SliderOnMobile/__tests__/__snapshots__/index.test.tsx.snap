// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderOnMobile Component SliderOnMobile should match snapshot 1`] = `
<div>
  <div
    class="container desktop-container"
  >
    <div>
      item1
    </div>
    <div>
      item2
    </div>
    <div>
      item3
    </div>
    <div>
      item4
    </div>
  </div>
  <div
    class="container mobile-container"
  >
    <div
      class="swiper-container swiper-container-initialized swiper-container-horizontal"
    >
      <div
        class="swiper-wrapper"
        style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="swiper-slide"
        >
          item1
        </div>
        <div
          class="swiper-slide"
        >
          item2
        </div>
        <div
          class="swiper-slide"
        >
          item3
        </div>
        <div
          class="swiper-slide"
        >
          item4
        </div>
      </div>
      <div
        class="swiper-pagination swiper-pagination-progressbar progress-bar"
      >
        <span
          class="progressbar-fill"
          style="webkit-transform: translate3d(0,0,0) scaleX(NaN) scaleY(1); transform: translate3d(0,0,0) scaleX(NaN) scaleY(1); webkit-transition-duration: 300ms; transition-duration: 300ms;"
        />
      </div>
      <span
        aria-atomic="true"
        aria-live="assertive"
        class="swiper-notification"
      />
    </div>
    <p
      class="count-slides"
    >
      01
       / 
      00
    </p>
  </div>
</div>
`;
